### PR TITLE
Add all positive cases for None and WildFly runtimes

### PR DESCRIPTION
**Pull request by Tasnim1147**

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -2,110 +2,411 @@ name: Nightly build
...
-    - cron: '0 0 * * *' # run at midnight UTC
+    - cron: "0 0 * * *" # run at midnight UTC
   workflow_dispatch:
 
 jobs:
   build-archetype:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Java SE 8
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:13]: Please restore this. We want to build using the lowest Java SE version. Alternatively, we can build with each Java SE version under test.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -2,110 +2,411 @@ name: Nightly build
...
           path: ~/.m2
-          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
-          restore-keys: ${{ runner.os }}-m2
+          key: ${ runner.os }-m2-${ hashFiles('**/pom.xml') }
+          restore-keys: ${ runner.os }-m2
 
       - name: Build Archetype
         run: mvn clean install --file archetype/pom.xml
 
-      - name: Run Archetype for EE 8, SE 8
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:29]: Why was this deleted? This is a legitimate test case.

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -2,110 +2,411 @@ name: Nightly build
...
+      - name: Set up Java SE 8
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 8
+
+      - name: Run Archetype for EE 9.1 Full Profile, SE 8, None
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:30]: Please fully describe each case. Is it using Docker or not? Also, no need to specify "none" - it is the default and should be  the first in terms of order.

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -2,110 +2,411 @@ name: Nightly build
...
+        run: |
+          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=full -DjavaVersion=8 -Druntime="none" -Ddocker=no -DoutputDirectory="app/none" -Dgoals="clean package"
+          rm -rf app/none
+
+      - name: Run Archetype for EE 9.1 Web Profile, SE 8, None
         run: |
-          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=web -DjavaVersion=8 -DoutputDirectory=app -Dgoals="clean package"
-          rm -rf app    
+          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=web -DjavaVersion=8 -Druntime="none" -Ddocker=yes -DoutputDirectory="app/none" -Dgoals="clean package"
+          rm -rf app/none
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:43]: No need to explicitly specify "none", it is the default case.

File: .github/workflows/nightly.yml [hunk 5]
```diff
@@ -2,110 +2,411 @@ name: Nightly build
...
 
-      - name: Run Archetype for EE 8, SE 8, GlassFish
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:39]: Please restore all needlessly deleted cases.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please address all review feedback.

File: .github/workflows/nightly.yml (hunk 3 above)
Comment (Tasnim1147) [reply .github/workflows/nightly.yml:30]: Will it be alright if the name is set in the following format:

For **None** runtime:
`- name: Run Archetype for EE {jakarta-version} {profile-name} Profile, SE {java-version}, with{out} docker`

For **other** runtimes:
`- name: Run Archetype for EE {jakarta-version} {profile-name} Profile, SE {java-version}, {runtime}, with{out} docker`

Comment (m-reza-rahman) [reply .github/workflows/nightly.yml:30]: Yes, except please capitalize Docker.

**Comment by m-reza-rahman** [comment]
Is this once again ready for review? If not, kindly set the status to draft until it is.

File: .github/workflows/nightly.yml (hunk 3 above)
Comment (Tasnim1147) [reply .github/workflows/nightly.yml:30]: Sure. Thank you for the clarification.

File: .github/workflows/nightly.yml
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:1]: Please order by SE version, runtime/none, EE version, profile, and Docker. Confusing to read otherwise.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please order by SE version (starting with the lowest), runtime/none (alphabetic, starting with none), EE version (starting with the lowest), profile (full, web, core), and Docker (no, yes).

So like this:

Run Archetype for EE 8 Web Profile, SE 8, Payara
Run Archetype for EE 8 Web Profile, SE 8, Payara, with Docker
Where Docker is really not supported (a warning is generated by the Archetype), we should not treat that as a "positive case". That's a negative case we need to test for later.

Please order your part and the file overall - including existing cases. Confusing to read and evaluate otherwise for me.

File: .github/workflows/nightly.yml [hunk 6]
```diff
@@ -30,82 +30,437 @@ jobs:
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=8 -Dprofile=full -DjavaVersion=8 -DoutputDirectory=app -Dgoals="clean package"
           rm -rf app
-          
+
+      - name: Run Archetype for EE 8 Full Profile, SE 8
```
Comment (m-reza-rahman) [inline .github/workflows/nightly.yml:34]: This is a duplicate of the one above. Please delete one.

**Review by m-reza-rahman — CHANGES_REQUESTED** [review]
Please re-read and understand the previous review comments. All cases where Docker is not a valid option should be removed as it is not really a "positive" case. These will be added later as negative cases where we explicitly check that a warning was generated and there is no Dockerfile.

**Comment by m-reza-rahman** [comment]
Is this ready for another review?

**Comment by m-reza-rahman** [comment]
NP. That’s the point of the review. Hopefully I can take a look during the work week.

**Comment by m-reza-rahman** [comment]
Looks great! Thanks for working through this.
